Enhancement: Add System.Environment::NewLine as a supported static property

### Description
MSBuild currently doesn't seem to provide the 'native' line ending for the current OS. (Did I miss a property or function that provides the line ending for the platform?)

There is a set of static methods and properties exposed including the `System.Environment::CommandLine` property (see [Static property functions](https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#static-property-functions)). Adding support for the `System.Environment::NewLine` property would provide the line ending for the current OS which would support greater portability for MSBuild scripts.

e.g., displaying an ItemGroup as one item per line could be written as:
```
<Message Text="@(stuff, '$([System.Environment]::NewLine)')" />
```

The example code currently generates an error:
```
error MSB4185: The function "NewLine" on type "System.Environment" is not available for execution as an MSBuild property function.
```

A less portable version of the example code is the following which hard-codes the \r\n for Windows:
```
<Message Text="@(stuff, '%0d%0a')" />
```

A more portable workaround would be:
```
<PropertyGroup
    <IsWindows>$([System.Runtime.InteropServices.RuntimeInformation]::IsOSPlatform($([System.Runtime.InteropServices.OSPlatform]::Windows)))</IsWindows>
    <Newline Condition="$(IsWindows)">%0d%0a</Newline>
    <Newline Condition="$(Newline) == ''">%0a</Newline>
</PropertyGroup>
<Message Text="@(stuff, '$(Newline)')" />
```

### Versions & Configurations
Checked on v16.10.1.58001

--- a/src/Build/Resources/Constants.cs
+++ b/src/Build/Resources/Constants.cs
@@ -301,12 +301,12 @@ namespace Microsoft.Build.Internal
                         availableStaticMethods.TryAdd("System.Environment::GetFolderPath", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::GetLogicalDrives", environmentType);
 
-// All the following properties only have getters
+                        // All the following properties only have getters
                         availableStaticMethods.TryAdd("System.Environment::CommandLine", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::Is64BitOperatingSystem", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::Is64BitProcess", environmentType);
-
                         availableStaticMethods.TryAdd("System.Environment::MachineName", environmentType);
+                        availableStaticMethods.TryAdd("System.Environment::NewLine", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::OSVersion", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::ProcessorCount", environmentType);
                         availableStaticMethods.TryAdd("System.Environment::StackTrace", environmentType);

